### PR TITLE
Search in Orders by Client Order Number

### PR DESCRIPTION
**Pull request by SebaKas21**

### CODE REVIEW

File: be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs [hunk 1]
```diff
@@ -163,12 +163,12 @@ public async Task<PagedResults<IEnumerable<OrderServiceModel>>> GetAsync(GetOrde
 
             if (model.IsSeller is false)
             {
-                orders = orders.Where(x => x.SellerId == model.OrganisationId.Value);
+                orders = orders.Where(x => x.SellerId == model.OrganisationId.Value || x.OrderItems.Any(y => y.ExternalReference.StartsWith(model.SearchTerm)) || x.Id.ToString() == model.SearchTerm);
```
Comment (dawiddworak88) [inline be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs]: **suggestion:** Please add ToLowercaseInvariant() to ignore the captial and lowercase in search.

File: be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs [hunk 2]
```diff
@@ -163,12 +163,12 @@ public async Task<PagedResults<IEnumerable<OrderServiceModel>>> GetAsync(GetOrde
...
             }
 
             if (!string.IsNullOrWhiteSpace(model.SearchTerm))
             {
-                orders = orders.Where(x => x.ClientName == model.SearchTerm || x.OrderItems.Any(y => y.ExternalReference == model.SearchTerm));
+                orders = orders.Where(x => x.ClientName.StartsWith(model.SearchTerm) || x.OrderItems.Any(y => y.ExternalReference.StartsWith(model.SearchTerm)) || x.Id.ToString() == model.SearchTerm);
```
Comment (dawiddworak88) [inline be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs]: suggestion: Please add ToLowercaseInvariant() to ignore the captial and lowercase in search.

**Review by dawiddworak88 — CHANGES_REQUESTED** [review]

**Comment by SebaKas21** [comment]
@dawiddworak88 

Now is okey?

File: be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs [hunk 3]
```diff
@@ -166,9 +166,11 @@ public async Task<PagedResults<IEnumerable<OrderServiceModel>>> GetAsync(GetOrde
                 orders = orders.Where(x => x.SellerId == model.OrganisationId.Value);
             }
 
-            if (!string.IsNullOrWhiteSpace(model.SearchTerm))
+            if (string.IsNullOrWhiteSpace(model.SearchTerm) is false)
             {
-                orders = orders.Where(x => x.ClientName == model.SearchTerm || x.OrderItems.Any(y => y.ExternalReference == model.SearchTerm));
+                orders = orders.Where(x => x.ClientName.ToLower().StartsWith(model.SearchTerm.ToLower())
```
Comment (dawiddworak88) [inline be/src/Project/Services/Ordering/Ordering.Api/Services/OrdersService.cs:171]: @SebaKas21 , now it's truly case insensitive. When order item has an external reference "ORD_01" and user searches for "ord", the right results are retrieved.

**Review by dawiddworak88 — APPROVED** [review]

**Pull request merged**